src/Lean/Meta/IndPredBelow.lean

```diff
@@ -303,10 +305,10 @@ where
         args.back.withApp fun ctor _ => do
         let ctorName := ctor.constName!.updatePrefix below.constName!
         let ctor := mkConst ctorName below.constLevels!
-        let ctorInfo ← getConstInfoCtor ctorName
-        let (mvars, _, _) ← forallMetaTelescope ctorInfo.type
-        let ctor := mkAppN ctor mvars
-        m.apply ctor
+        let ms ← m.apply ctor
+        trace[Meta.IndPredBelow] "new goals {ms}"
+        pure ms.reverse
```

Comment:
This change broke some other tests, until I added the `reverse` here; now the existing tests succeed. :shrug:

Note to my future self:

I wonder if this just means that for any order of mvars we’ll have examples where it fails, or if there is really a reason why this order is good™. I’ll have to come back to this and think about it some more.

Maybe I’ll also investigate if we avoid proof search here and build the necessary terms direct, relying on the structure of the `.ibelow` construction.